src/state/dimension.js

```diff
@@ -136,8 +143,12 @@ export const getDroppableDimension = ({
 }: GetDroppableArgs): DroppableDimension => {
   const withMargin = getWithSpacing(clientRect, margin);
   const withWindowScroll = getWithPosition(clientRect, windowScroll);
-  // If no containerRect is provided, the clientRect is the container
-  const containerRectWithWindowScroll = getWithPosition(containerRect || clientRect, windowScroll);
+  // If no containerRect is provided, or if the clientRect matches the containerRect, this
+  // droppable is its own container. In that case we include margin in the container dimension.
+  const droppableIsContainer = !containerRect || equal(containerRect, clientRect);
+  const containerRectWithWindowScroll = droppableIsContainer
+    ? getWithPosition(withMargin, windowScroll)
+    : getWithPosition(containerRect || withMargin, windowScroll);
```

Comment:
can we ever have not have a containerRect when the droppable is not the container?

Reply:
no, but flow was having a sad about it 😕 I might re-write the logic so flow realises that it's impossible for containerRect to be undefined at this point.

Reply:
Yeah, perhaps just add a line that throws if clientRect is null. It would be clearer then :)